lib/ansible/modules/cloud/amazon/aws_ses_identity_dkim.py

```diff
@@ -0,0 +1,223 @@
+      the identity tokens for you to add to DNS.
+version_added: "2.9"
+author: Stefan Horning (@stefanhorning)
+
+options:
+    identity:
+        description:
+            - The SES identity you would like to enable DKIM for.
+            - This should be the email address or domain identifying an existing SES identity
+        type: str
+        required: true
+    state:
+        description:
+            - Enable or disable DKIM on the identity
+        default: enabled
+        choices: [ 'enabled', 'disabled']
+        type: str
+requirements: [ 'botocore', 'boto3' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: to setup SES identities use the aws_ses_identity module.
+
+- name: Enable DKIM setting for identity example@example.com
+  aws_ses_identity_dkim:
+      identity: example@example.com
+
+- name: Enable DKIM setting for identity example.com
+  aws_ses_identity_dkim:
+    identity: example.com
+  register: dkim_results
+
+- name: Example how to aws_ses_identity_dkim results to set DNS records for validation (if you use Route53 service for DNS)
+  route53:
+    record: "{{ item }}._domainkey.example.com"
+    value: "{{ item }}.dkim.amazonses.com"
+    type: CNAME
+    zone: example.com
+  with_items: "{{ dkim_results.dkim_attributes.dkim_tokens }}"
+
+- name: Disable DKIM setting for identity example.com
+  aws_ses_identity_dkim:
+    identity: example.com
+    state: disabled
+'''
+
+RETURN = '''
+dkim_attributes:
+    description: Dictionary with DKIM information
+    type: complex
+    returned: always
+    sample: {
+        "dkim_tokens": [],
+        "dkim_enabled": "True",
+        "dkim_verification_status": "Pending"
+    }
+    contains:
+        dkim_tokens:
+            descritpion: Array of the 3 DKIM verification tokens needed for DNS records
+            type: list
+            sample: ['u5d5fa5twxzdni4aaoffdqvgnvku5kvu', '6xzw7n3abimypkyan5s6kuse4nymrfo2', 'nkwigx63eervm4hyja2lwzhjcfqyhka7']
+        dkim_enabled:
+            description: Boolean indicating if DKIM email sending is enabled or disabled for identity
+            type: bool
+            sample: True
+        dkim_verification_status:
+            description: Status of the DKIM verification process (DNS verification)
+            type: str
+            sample: 'Pending'
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, AWSRetry
+
+import os
+import time
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+def _get_state_check(state):
+    if state == 'enabled':
+        return lambda attributes: attributes['DkimEnabled'] and attributes['DkimTokens']
+    elif state == 'disabled':
+        return lambda attributes: not attributes['DkimEnabled']
+    elif state == 'any':
+        return lambda attributes: True
```

Comment:
OR somehow catch key errors here and return s.th. instead...

Reply:
Not quite sure what you mean here, can you expand a bit?

What I'm trying to do with the any case is to allow for a "don't wait" case in get_dkim_attributes when we're not waiting for a particular state without having a special case. But it may be clearer to just say wait_for=None means don't wait and get rid of the any case here.

Reply:
If we don't have the key `DkimEnabled` (see `dkim_attributes comment` above) we get a key error here.

Reply:
I get it. I think the answer is actually to leave this unchanged and make get_identity_dkim_attributes always call the real method even if we're in check mode (see comment on that method).